autotest/ogr/ogr_pgeo.py

```diff
@@ -742,3 +742,39 @@ def test_ogr_pgeo_read_metadata():
 
     ds.ReleaseResultSet(sql_lyr)
 
+
+###############################################################################
+# Test retrieving layer names with non-ascii characters
+
+@pytest.mark.skipif(sys.platform == 'win32', reason="See https://github.com/OSGeo/gdal/issues/2894")
+def test_ogr_pgeo_layer_name_encoding():
+    ds = gdal.OpenEx('data/pgeo/encoding.mdb', gdal.OF_VECTOR)
+
+    assert ds.GetLayerCount() == 2, 'did not get expected layer count'
+
+    assert [ds.GetLayer(i).GetName() for i in range(2)] == ['special_chars_áõã', 'táble']
```

Comment:
Maybe a bit risky to assume that the layer order will be stable
```suggestion
    assert set([ds.GetLayer(i).GetName() for i in range(2)]) == set(['special_chars_áõã', 'táble'])
```